src/components/sdfgViewer.ts

```diff
@@ -11,7 +11,41 @@ import { AnalysisProvider } from './analysis';
 import { BaseComponent } from './baseComponent';
 import { ComponentMessageHandler } from './messaging/componentMessageHandler';
 import { TransformationListProvider } from './transformationList';
-import { BreakpointHandler, getCppRange, Node } from '../debugger/breakpointHandler';
+import { getCppRange, Node } from '../debugger/breakpointHandler';
+
+class Message {
+    timeStamp: Date;
+    sdfgName: string;
+    message: any;
+
+    constructor(sdfgName: string, message: any) {
+        this.sdfgName = sdfgName;
+        this.timeStamp = new Date();
+        this.message = message;
+    }
+
+    public checkTimestamp(): boolean {
+        // Checks if the msg isn't to old. We give the webview
+        // 30s to handle a msg otherwise its outdated
+        return new Date().getTime() < this.timeStamp.getTime() + 30000;
+    }
+
+    public sendMessage() {
+        DaCeVSCode.getInstance().getActiveEditor()?.postMessage(this.message);
+    }
+
+    public executeMessage(sdfgName: string): boolean {
+        // Sends the msg if it corresponds to the SDFG and is not outdated
+        // Returns true if the Message can be deleted
+        if (sdfgName === this.sdfgName) {
```

Comment:
Shouldn't the SDFG name always match in any given instance of `SdfgViewer`? If not, consider the fact that a user may have multiple SDFGs open at the same time, all named 'program.sdfg' under different paths - how would this behave in that situation?

Reply:
The sdfgName is the name of the SDFG (sdfg.attributes.name) and not the filename. However, in the unlikely but possible situation that 2 SDFGs are getting opened (by the command openViewer) at almost the same time with the same SDFG name, messages could be handled wrongly. To mitigate this we would have to also compare the file path. The problem is, to be able to do that the webview would have to send the file path of its opened sdfg which it doesn't have access to as it runs in a protected environment.

Reply:
That makes sense - in that case let's leave it as is for now, I agree that the case is very unlikely to ever occurr